packages/asgardex-binance/src/client.ts

```diff
@@ -155,158 +174,245 @@ class Client implements BinanceClient {
+          to: transaction.to,
+          coins: transaction.coins.map(coin => {
+            return {
+              denom: coin.asset.symbol,
+              amount: baseAmountToBig(coin.amount)
+            }
+          })
+        }
+      }),
+      memo
+    )
 
-    const from = addressFrom || this.getAddress()
-    if (!from)
-      return Promise.reject(
-        new Error(
-          'Parameter `addressFrom` has to be set. Or set a phrase by calling `setPhrase` before to use an address of an imported key.',
-        ),
-      )
-    return await this.bncClient.transfer(from, addressTo, amount, asset, memo)
+    try {
+      return transferResult.result.map((txResult: any) => txResult.hash)[0]
+    } catch (err) {
+      return ''
+    }
   }
 
-  normalTx = async ({ addressFrom, addressTo, amount, asset }: NormalTxParams): Promise<TransferResult> => {
+  deposit = async ({ asset, amount, recipient, memo }: TxParams): Promise<TxHash> => {
+    return this.transfer({ asset, amount, recipient, memo })
+  }
+
+  transfer = async ({ asset, amount, recipient, memo }: TxParams): Promise<TxHash> => {
     await this.bncClient.initChain()
-    await this.setPrivateKey().catch((error) => Promise.reject(error))
+    await this.bncClient.setPrivateKey(this.getPrivateKey()).catch((error) => Promise.reject(error))
 
-    const from = addressFrom || this.getAddress()
-    if (!from)
-      return Promise.reject(
-        new Error(
-          'Parameter `addressFrom` has to be set. Or set a phrase by calling `setPhrase` before to use an address of an imported key.',
-        ),
-      )
-    return await this.bncClient.transfer(from, addressTo, amount, asset)
+    const transferResult = await this.bncClient.transfer(
+      this.getAddress(),
+      recipient,
+      amount.amount().div(1e8).toNumber(),
+      asset.symbol,
+      memo
+    )
+
+    try {
+      return transferResult.result.map((txResult: any) => txResult.hash)[0]
+    } catch (err) {
+      return ''
+    }
   }
 
-  freeze = async ({ address, asset, amount }: FreezeParams): Promise<TransferResult> => {
+  freeze = async ({ recipient, asset, amount }: FreezeParams): Promise<TxHash> => {
     await this.bncClient.initChain()
-    await this.setPrivateKey().catch((error) => Promise.reject(error))
+    await this.bncClient.setPrivateKey(this.getPrivateKey()).catch((error) => Promise.reject(error))
 
-    const addr = address || this.getAddress()
-    if (!addr)
+    const address = recipient || this.getAddress()
+    if (!address)
       return Promise.reject(
         new Error(
           'Address has to be set. Or set a phrase by calling `setPhrase` before to use an address of an imported key.',
         ),
       )
-    return await this.bncClient.tokens.freeze(addr, asset, amount)
+
+    const transferResult = await this.bncClient.tokens.freeze(address, asset.symbol, baseAmountToBig(amount))
+
+    try {
+      return transferResult.result.map((txResult: any) => txResult.hash)[0]
```

Comment:
Pls try to avoid `any`. Are your sure `hash` is available in `txResult`. Just because documentation about [multiSend](https://github.com/binance-chain/javascript-sdk/blob/32d09760656915f8cdbebffcf3cee940442f4080/docs/api-docs/classes/bncclient.md#multisend) does not give us a `hash` nor [multiSend](https://github.com/binance-chain/javascript-sdk/blob/916cfd83b488a9404d1a4820a599d9ca8c28a002/src/client/index.ts#L410-L439)

However, we could do something like that to avoid `any` here:

```
return transferResult.result.map((txResult: { hash?: string }) => txResult?.hash ?? '')[0]
```